processing/src/main/java/org/apache/carbondata/processing/loading/model/CarbonLoadModelBuilder.java

```diff
@@ -426,6 +432,24 @@ private void validateAndSetColumnCompressor(CarbonLoadModel carbonLoadModel)
     }
   }
 
+
+  private void validateAndSetBinaryDecoder(CarbonLoadModel carbonLoadModel)
+      throws InvalidLoadOptionException {
+    try {
+      String binaryDecoder = carbonLoadModel.getBinaryDecoder();
+      if (StringUtils.isBlank(binaryDecoder)) {

```

Comment:
No need to set default as current default is empty decoder.
If the configured binaryDecoder is non empty, then validate with allowed values "hex", "base64"

Reply:
it should set current value  or set default value if didn't set hex and base64 in here, otherwise it will throw exception:
```
DataLoad failure: null
java.lang.Exception: DataLoad failure: null
	at org.apache.carbondata.spark.rdd.CarbonDataRDDFactory$.loadCarbonData(CarbonDataRDDFactory.scala:491)
	at org.apache.spark.sql.execution.command.management.CarbonLoadDataCommand.loadData(CarbonLoadDataCommand.scala:630)
	at org.apache.spark.sql.execution.command.management.CarbonLoadDataCommand.processData(CarbonLoadDataCommand.scala:357)
	at org.apache.spark.sql.execution.command.AtomicRunnableCommand$$anonfun$run$3.apply(package.scala:148)
	at org.apache.spark.sql.execution.command.AtomicRunnableCommand$$anonfun$run$3.apply(package.scala:145)
	at org.apache.spark.sql.execution.command.Auditable$class.runWithAudit(package.scala:104)
	at org.apache.spark.sql.execution.command.AtomicRunnableCommand.runWithAudit(package.scala:141)
	at org.apache.spark.sql.execution.command.AtomicRunnableCommand.run(package.scala:145)
```